sympy/printing/fcode.py

```diff
@@ -178,6 +172,32 @@ def _print_Piecewise(self, expr):
                                       "inline operators is not supported in "
                                       "standards earlier than Fortran95.")
 
+    def _print_Assignment(self, expr):
```

Comment:
Well, since upwind in this case means Assignment, e.g.
`Assignment(Symbol('x'), sign(Symbol('x')))`
(the printer walks the tree and sees assignment before sing, and then the printing method of `Assignment` has a special case for `Piecewise` and now also `sign`). Therefore we will still need a `_print_sign` method for the cases when it is not part of an assignment. But I can refactor so that `_print_Assignment` and `_print_sign` share some private function with the logic for printing `sign` using `Piecewise`